BASE/src/ServerTelemetryChannel/ServerTelemetryChannel.cs

```diff
@@ -243,6 +246,21 @@ public int LocalThrottleWindow
             set { this.Transmitter.ThrottleWindow = value; }
         }
 
+        /// <summary>
+        /// Gets or sets the <see cref="CredentialEnvelope"/> which is used for AAD.
+        /// FOR INTERNAL USE. Customers should use <see cref="TelemetryConfiguration.SetAzureTokenCredential"/> instead.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="ServerTelemetryChannel.CredentialEnvelope"/> on sets <see cref="Transmitter.CredentialEnvelope"/> and then sets <see cref="TransmissionSender.CredentialEnvelope"/> 
+        /// which is used to set <see cref="Transmission.CredentialEnvelope"/> just before calling <see cref="Transmission.SendAsync"/>.
+        /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public CredentialEnvelope CredentialEnvelope
```

Comment:
this needs to be public?

Reply:
Either the Interface and Property are public.
Or the Interface is private, and the properties have to be explicitly implemented. 

I don't have a strong opinion on which implementation, but I need to be able to set this property in `ServerTelemetryChannel` from the `TelemetryConfiguration`

Reply:
Keep the interface internal (with InternalsVisibleTo for the Microsoft.AI.ServerTelemetryChannel assembly) and use explicit interface implementation.
It's then impossible for user code to call it by mistake (without reflection).
You don't even need the EditorBrowsable attribute.
(You have in right in your private branch!)